web/packages/teleport/src/lib/useMfa.ts

```diff
@@ -172,7 +172,7 @@ export function useMfa({ req, isMfaRequired }: MfaProps): MfaState {
   };
 }
 
-export function useMfaTty(emitterSender: EventEmitterMfaSender): MfaState {
+export function useMfaEmitter(emitterSender: EventEmitterMfaSender): MfaState {
```

Comment:
Something still feels off about this. 

Are you sure desktop code should be calling this? AFAIK, the TermEvent enum is a bunch of events that our web SSH console depends on. I wouldn't expect the desktop player to depend on any term events since it is not a terminal. 

If your intent is some refactoring so that SSH and desktop MFA share more code I think that's a fine goal, but I think the abstraction is probably not quite right. I don't think desktop code should depend on TermEvent.

Reply:
I didn't make any significant refactor to the Desktop session MFA code, it was already using `useMfa`, which took `Tty` or `EventEmitterMfaSender` as an argument (`tdpClient` implements `EventEmitterMfaSender`). I renamed it to `useMfaTty` or `useMfaEmitter` so that I could make `useMfa` more generally usable. 

See the definition for Tty in the WebUI: `class Tty extends EventEmitterMfaSender`. This just represents the websocket used in WebUI sessions to send MFA challenges from the proxy and retrieve a challenge response from the client. I can take a stab at renaming to make this more clear.

Reply:
@avatus Can you help provide any more clarification? It is a dense section of the code base.

Reply:
I think this is a result of poor naming/abstractions from the past. desktop and ssh docs both implemented the old `EventWebAuthnEmitter` or whatever it was called, which had a single non-implemented `sendMfa` method that tty and tdpClient had to implement themselves. And we've since updated it from `useWebauthn` to `useMfa` but that weird abstraction layer should probably just go imo. Outside of this PR probably